### PR TITLE
add devdocs how to profile package precompilation with tracy

### PR DESCRIPTION
**Pull request by KristofferC**

### CODE REVIEW

**Comment by topolarity** [comment]
Should we make `precompile_wait_tracy` self-resetting, so that it doesn't make future pre-compiles hang (especially those triggered by `using`)?

**Comment by KristofferC** [comment]
> Should we make precompile_wait_tracy self-resetting, so that it doesn't make future pre-compiles hang (especially those triggered by using)?

Yes, seems like a good idea.

File: base/Base.jl [hunk 1]
```diff
@@ -596,6 +596,7 @@ function __init__()
         ccall(:jl_set_peek_cond, Cvoid, (Ptr{Cvoid},), PROFILE_PRINT_COND[].handle)
         errormonitor(Threads.@spawn(profile_printing_listener()))
     end
+    delete!(ENV, "JULIA_WAIT_FOR_TRACY")
```
Comment (topolarity) [inline base/Base.jl:599]: Might be nice to leave a note here about why we're messing with `ENV`

**Review by topolarity — APPROVED** [review]
Quite happy with this latest version :+1:

Thanks!

**Review by topolarity — APPROVED** [review]

**Pull request merged**